modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3ClientSettings.java

```diff
@@ -536,5 +560,6 @@ static final class Defaults {
         static final TimeValue CONNECTION_MAX_IDLE_TIME = TimeValue.timeValueSeconds(60);
         static final int MAX_CONNECTIONS = 50;
         static final int RETRY_COUNT = 3;
+        static final TimeValue API_CALL_TIMEOUT = TimeValue.ZERO; // default to no API call timeout
```

Comment:
Could we use `TimeValue.MINUS_ONE` to mean "no timeout", leaving `TimeValue.ZERO` to mean "time out immediately"? Or at least to defer the meaning of zero down to the SDK, I haven't followed through to determine whether zero means zero or infinity within the SDK but I'd prefer we had some way to say "don't even invoke `clientOverrideConfiguration.apiCallTimeout`".

Reply:
I thought the SDK treats `0` as no timeout after reading the [TimeoutTracker](https://github.com/aws/aws-sdk-java-v2/blob/57a68453d6eecadc7e18ae8bd5448332ade61cf7/core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/timers/TimerUtils.java#L78-L79) code. Didn't know it performs validations earlier on. Now fixed in 4e3c012853e880288820a30e54dd09fa6fe102e8